python/tvm/tir/function.py

```diff
@@ -239,3 +240,40 @@ def get(name: str):
             The TensorIntrin with the specified name.
         """
         return _ffi_api.TensorIntrinGet(name)  # pylint: type: ignore
+
+
+@tvm._ffi.register_object("tir.IndexMap")
+class IndexMap(Object):
+    """A mapping from multi-dimensional indices to another set of multi-dimensional indices
+
+    Parameters
+    ----------
+    initial_indices : List[Var]
+        Variables representing the indices prior to remapping.
+    final_indices : List[PrimExpr]
+        Expressions defining the indices after remapping.
+    """
+
+    initial_indices: List[Var]
+    final_indices: List[PrimExpr]
+
+    @staticmethod
+    def from_func(func: Callable) -> "IndexMap":
```

Comment:
updated, for (1), I changed this to call constructor in python side, because even if we propagate the names, there are still naming issues of using `IndexMap::FromFunc` in C++ side

Reply:
Looks good!  I like having the `mapping_function` be handled primarily on the Python side, because it avoids the need for a round-trip Python -> C++ -> Python call in order to generate the mapping, and feels more idiomatic to have it explicitly specified on the C++ side.